File: 1. Programming and Programming Tools/Hour Long Code Practise/Life.java
/*
* Codify my Life
*
*/

import java.util.Map;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

class Life{
	private static final String TODAY = "Monday";
	private static enum DAY_OF_WEEK {Monday,Tuesday,Wednesday,Thursday,Friday};
	private static final String[] BIRTHDAY = {"1955","01","01"};
	private static final Map<String,Boolean> IS_WEEKEND= new HashMap<>();
	private static final Set<String> IS_VEGETARIAN_DAY = new HashSet<>();
	
	public static void main(String[] args){
		System.out.println("This is my life");
		//Read the day of week from the args, else ask the user for the day of week
		for(int i=0;i<3;i++){
			switch(i){
				case 0: System.out.println("Year: "+BIRTHDAY[i]);
				break;
				case 1: System.out.println("Month: "+BIRTHDAY[i]);
				break;
				case 2: System.out.println("Day: "+BIRTHDAY[i]);
				break;
				default: System.out.println("Birthday is not correct");
			}
			
		}
		checkIfWeekend(TODAY);

		IS_VEGETARIAN_DAY.add("Monday");
		checkIfVegetarian(TODAY);
	}


	public static void checkIfWeekend(String today){
		IS_WEEKEND.put("Saturday",true);
		IS_WEEKEND.put("Sunday",true);

		if(IS_WEEKEND.containsKey(TODAY)){
			if(IS_WEEKEND.get(TODAY)){
				System.out.println("Yay today is weekend");
			}
		}else{
			System.out.println("Today is a weekday");
		}
	}

	public static void checkIfVegetarian(String today){
		if(IS_VEGETARIAN_DAY.contains(today)){
			System.out.println("Today is vegetarian");
		}else{
			System.out.println("Today is non vegetarian");
		}
	}
}

File: 1. Programming and Programming Tools/Hour Long Code Practise/Life.py
class Life:
	@staticmethod
	def run():
		TODAY="Monday"
		DAYS = ("Mon","Tue","Wed","Thu","Fri")
		BIRTHDAY = ("1955","01","01")
		print("This is my life")
		print("Today is " + DAYS[0])

		for i in range(0,3):
			if i==0 :
				print("Year is "+ BIRTHDAY[i])
			elif i==1:
				print("Month is "+ BIRTHDAY[i])
			elif i==2:
				print("Day is "+ BIRTHDAY[i])
		Life.checkIfWeekend(TODAY)

	@staticmethod
	def checkIfWeekend(today):
		IS_WEEKEND = {"Saturday":True, "Sunday":True}
		if today in IS_WEEKEND:
			if IS_WEEKEND[today] == True:
				print("Today is a weekend")
		else:
			print("Today is a weekday")

Life.run()
